pkg/R/RDD.R

```diff
@@ -1374,6 +1374,42 @@ setMethod("top",
             takeOrderedElem(rdd, num, FALSE)
           })
 
+#' Pipes elements to a forked externel process.
+#'
+#' The same as `pipe()' in Spark.
+#'
+#' @param rdd the RDD whose elements are piped to the forked externel process.
+#' @param command the command to fork an externel process.
+#' @param ... other arguments passed to pipeRDD.
+#' @param env character vector of name=value strings to set environment variables.
+#' @return a new RDD created by piping all elements to a forked externel process.
+#' @rdname pipeRDD
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, 1:10)
+#' collect(pipeRDD(rdd, "cat")
+#' Output: c("1", "2", ..., "10")
+#'}
+setGeneric("pipeRDD", function(rdd, command, ...) { standardGeneric("pipeRDD") })
+
+#' @rdname pipeRDD
+#' @aliases pipeRDD,RDD,character-method
+setMethod("pipeRDD",
+          signature(rdd = "RDD", command = "character"),
+          function(rdd, command, env = character()) {
+            func <- function(part) {
+              trim.trailing.func <- function(x) {
```

Comment:
The function is used to trim the input fed to forked process so that system2 will not mess up the records by treating "\n" as a new line. Actually, I think only "\n" and "\r\n" needs to be removed.